src/python/pants/engine/internals/graph.py

```diff
@@ -215,7 +204,7 @@ async def resolve_target_parametrizations(
     target_types_to_generate_requests: TargetTypesToGenerateTargetsRequests,
     files_not_found_behavior: FilesNotFoundBehavior,
 ) -> _TargetParametrizations:
-    assert not address.is_generated_target and not address.is_parametrized
+    assert not address.is_generated_target
```

Comment:
This is the only divergance from pre-https://github.com/pantsbuild/pants/pull/14408. A target generator _can_ be parametrized, as shown in the PR description + tests.